File: .skills/openclaw-skills/skills/ai-hub-admin/craftsman-agent-3d-generation/scripts/generate_minecraft_build_plan.ts
#!/usr/bin/env node

const ENDPOINT = "https://agent.deepnlp.org/agent";
const UNIQUE_ID = "craftsman-agent/craftsman-agent";
const API_ID = "generate_minecraft_build_plan";
const ENV_KEY = "DEEPNLP_ONEKEY_ROUTER_ACCESS";
const DEMO_KEY = "BETA_TEST_KEY_MARCH_2026";

function sleep(ms: number): Promise<void> {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

function parseArgs(argv: string[]) {
  const args: Record<string, string | string[]> = {};
  for (let i = 0; i < argv.length; i += 1) {
    const token = argv[i];
    if (token === "--prompt" || token === "--mode" || token === "--ref-image-url") {
      const value = argv[i + 1];
      i += 1;
      if (token === "--ref-image-url") {
        const list = (args[token] as string[]) || [];
        list.push(value);
        args[token] = list;
      } else {
        args[token] = value;
      }
    }
  }

  return {
    prompt: (args["--prompt"] as string) || "",
    mode: (args["--mode"] as string) || "basic",
    refImageUrl: (args["--ref-image-url"] as string[]) || [],
  };
}

async function main() {
  const { prompt, mode, refImageUrl } = parseArgs(process.argv.slice(2));
  if (!prompt) {
    console.error("Missing --prompt");
    process.exit(1);
  }

  let apiKey = process.env[ENV_KEY];
  if (!apiKey) {
    console.error(
      "DEEPNLP_ONEKEY_ROUTER_ACCESS is not set. The API is not free; using demo key after a short wait."
    );
    console.error("Set with: export DEEPNLP_ONEKEY_ROUTER_ACCESS=YOUR_API_KEY");
    await sleep(2000);
    apiKey = DEMO_KEY;
  }

  const url = new URL(ENDPOINT);
  url.searchParams.set("onekey", apiKey);

  const payload = {
    unique_id: UNIQUE_ID,
    api_id: API_ID,
    data: {
      prompt,
      ref_image_url: refImageUrl,
      mode,
    },
  };

  const response = await fetch(url.toString(), {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });

  const text = await response.text();
  if (!response.ok) {
    console.error(`HTTP ${response.status}: ${text}`);
    process.exit(1);
  }

  try {
    const json = JSON.parse(text);
    console.log(JSON.stringify(json, null, 2));
  } catch {
    console.log(text);
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: .skills/openclaw-skills/skills/ai-hub-admin/craftsman-agent-3d-generation/scripts/generate_minecraft_build_plan.py
#!/usr/bin/env python3
"""Call Craftsman Agent OneKey Router to generate a Minecraft build plan."""

import argparse
import json
import os
import sys
import time
from urllib import request, parse, error

ENDPOINT = "https://agent.deepnlp.org/agent"
UNIQUE_ID = "craftsman-agent/craftsman-agent"
API_ID = "generate_minecraft_build_plan"
ENV_KEY = "DEEPNLP_ONEKEY_ROUTER_ACCESS"
DEMO_KEY = "BETA_TEST_KEY_MARCH_2026"


def warn_missing_key():
    sys.stderr.write(
        "DEEPNLP_ONEKEY_ROUTER_ACCESS is not set. "
        "The API is not free; using demo key after a short wait.\n"
    )
    sys.stderr.write(
        "Set with: export DEEPNLP_ONEKEY_ROUTER_ACCESS=YOUR_API_KEY\n"
    )
    time.sleep(2)


def build_payload(prompt, ref_image_url, mode):
    return {
        "unique_id": UNIQUE_ID,
        "api_id": API_ID,
        "data": {
            "prompt": prompt,
            "ref_image_url": ref_image_url,
            "mode": mode,
        },
    }


def post_json(url, payload):
    data = json.dumps(payload).encode("utf-8")
    req = request.Request(
        url,
        data=data,
        headers={"Content-Type": "application/json"},
        method="POST",
    )
    with request.urlopen(req, timeout=60) as resp:
        body = resp.read().decode("utf-8")
        return resp.status, body


def main():
    parser = argparse.ArgumentParser(description="Generate Minecraft build plan via Craftsman Agent")
    parser.add_argument("--prompt", required=True, help="Text prompt for the build")
    parser.add_argument(
        "--ref-image-url",
        action="append",
        default=[],
        help="Reference image URL (repeatable)",
    )
    parser.add_argument("--mode", default="basic", help="demo|basic|standard|advanced")
    args = parser.parse_args()

    api_key = os.getenv(ENV_KEY)
    if not api_key:
        warn_missing_key()
        api_key = DEMO_KEY

    query = parse.urlencode({"onekey": api_key})
    url = f"{ENDPOINT}?{query}"

    payload = build_payload(args.prompt, args.ref_image_url, args.mode)

    try:
        status, body = post_json(url, payload)
    except error.HTTPError as e:
        error_body = e.read().decode("utf-8") if e.fp else ""
        sys.stderr.write(f"HTTP {e.code}: {error_body}\n")
        sys.exit(1)
    except error.URLError as e:
        sys.stderr.write(f"Network error: {e.reason}\n")
        sys.exit(1)

    try:
        parsed = json.loads(body)
        print(json.dumps(parsed, indent=2))
    except json.JSONDecodeError:
        print(body)


if __name__ == "__main__":
    main()
